### PR TITLE
fix: accept missing mix hashes in remote blocks

### PR DESCRIPTION
**Pull request by fvictorio**
See https://github.com/NomicFoundation/hardhat/discussions/5266#discussioncomment-9776377

I didn't add a test for this because neither Infura nor Alchemy support that creditcoin chain, and I don't think we should use public RPC URLs in our tests, since they don't tend to be very robust.

### CODE REVIEW

File: crates/edr_evm/src/block/remote.rs [hunk 1]
```diff
@@ -79,7 +79,7 @@ impl RemoteBlock {
             gas_used: block.gas_used,
             timestamp: block.timestamp,
             extra_data: block.extra_data,
-            mix_hash: block.mix_hash.ok_or(CreationError::MissingMixHash)?,
+            mix_hash: block.mix_hash.unwrap_or_default(),
```
Comment (Wodann) [inline crates/edr_evm/src/block/remote.rs:82]: I think we should add a GitHub issue and `TODO` here to indicate that this should be reverted once we have multi-chain support.

Then we (or someone else) can add an implementation for this particular chai.

**Review by Wodann — CHANGES_REQUESTED** [review]

**Review by Wodann — APPROVED** [review]
Otherwise LGTM

**Review by agostbiro — APPROVED** [review]
Thanks LGTM

**Comment by jeremyjams** [comment]
Thank you for the fix! Do you have any idea when you plan to release it?
When EDR is released, would [going through this "process"](https://github.com/NomicFoundation/hardhat/pull/5087#issuecomment-2042188903) be enough to benefit from this fix from a fresh hardhat dependency?

File: crates/edr_evm/src/block/remote.rs (hunk 1 above)
Comment (fvictorio) [reply crates/edr_evm/src/block/remote.rs]: Done, thanks.

Can you re-review this? I had to solve a merge conflict and it was pretty straightforward, but it wouldn't hurt to double-check.

**Comment by fvictorio** [comment]
@jeremyjams I guess we'll merge this today, and might release it later this week.

> When EDR is released, would https://github.com/NomicFoundation/hardhat/pull/5087#issuecomment-2042188903 be enough to benefit from this fix from a fresh hardhat dependency?

Correct.

**Review by Wodann — APPROVED** [review]

**Comment by jeremyjams** [comment]
Hi, do you have any update on that?

**Comment by jeremyjams** [comment]
Hello @fvictorio no problem, thank you for this release and your previous message :pray: (I'm sadly now facing another issue https://github.com/NomicFoundation/edr/issues/536)

**Pull request merged**